File: JavaSource/src/main/java/ru/akhcheck/patterns/command/commands/CreateCocktailCommand.java
package ru.akhcheck.patterns.command.commands;

import java.util.Map;
import ru.akhcheck.patterns.command.cookery.Bartender;

/** Команда на приготовление коктейля. */
public class CreateCocktailCommand implements Command {
    private static final Map<String, Integer> PRICES = Map.of("Mojito", 480, "Americano", 200);

    private final Bartender bartender;
    private final String drink;

    public CreateCocktailCommand(Bartender bartender, String drink) {
        this.bartender = bartender;
        this.drink = drink;
    }

    @Override
    public void execute() {
        bartender.mix(drink);
    }

    @Override
    public int getPrice() {
        return PRICES.getOrDefault(drink, 0);
    }

    @Override
    public String toString() {
        return "cocktail: " + drink;
    }
}

File: Command/python-source/commands/CreateCocktailCommand.py
from __future__ import annotations

import sys
from pathlib import Path
from typing import ClassVar

if str(Path(__file__).resolve().parents[1]) not in sys.path:
    sys.path.append(str(Path(__file__).resolve().parents[1]))

from cookery import Bartender

from .Command import Command


class CreateCocktailCommand(Command):
    PRICES: ClassVar[dict[str, int]] = {"Mojito": 480, "Americano": 200}

    def __init__(self, bartender: Bartender, drink: str) -> None:
        self._bartender = bartender
        self._drink = drink

    def execute(self) -> None:
        self._bartender.mix(self._drink)

    def get_price(self) -> int:
        return self.PRICES.get(self._drink, 0)

    def __str__(self) -> str:
        return f"cocktail: {self._drink}"
